### ISSUE TITLE
Color attribute for materials/cells/universes

### ISSUE THREAD

**Issue by pshriwise** [openmc-dev/openmc#2506]
Similar to #2503, I have a plotting query for the community.

The assignment of material/cell colors in a dictionary when creating a plotting object has always felt a little clunky to me. Serpent allows users to associate a color with a cell or material that will be applied in any plots or visualization. At first, it felt odd to me to mix something non-physical into the object description, but having the color assigned to the object _would_ make it straightforward to keep colors consistent in plots generated by the Python API, the openmc executable, and the [plotting utility](https://github.com/openmc-dev/plotter).

Thoughts @paulromano @kkiesling @gridley?

**Comment by kkiesling** [comment]
I agree this would be a very useful feature. I assume this would be added as an attribute on the material or cell object that could be read across these different utilities.

**Comment by yrrepy** [comment]
Yes, I've recently discovered there was the Python API plotting, it's a bit confusing having 3 diff. ways to plot, but I guess each has it's merit.
It'd be nice to keep things more constant/consistent across the three methods.

**Comment by gridley** [comment]
Sounds nice, but I think rather than putting a color attribute on the cell or material class, it would be better to keep two hashmaps somewhere that map cell/material IDs into color values. That way we're not inflating the memory consumption of large models at all.

**Comment by pshriwise** [comment]
@gridley We should definitely keep large models in mind for cell/material attributes 👍🏻  I'll keep that in mind should I find some time to submit a PR for this soon.

**Comment by tjlaboss** [comment]
This would be a nice feature! I've been guilty of trying to set `Material.color` more than once.

**Comment by tjlaboss** [comment]
Partly addressed by PR #2826 (Material and Cell only, not Universe)
